tls/s2n_config.c

```diff
@@ -986,3 +986,10 @@ int s2n_config_client_hello_cb_enable_poll(struct s2n_config *config) {
 
     return S2N_SUCCESS;
 }
+
+int s2n_config_set_send_buffer_size(struct s2n_config *config, uint32_t size) {
+    POSIX_ENSURE_REF(config);
+    POSIX_ENSURE(size > S2N_TLS_MAX_RECORD_LEN_FOR(0), S2N_ERR_INVALID_ARGUMENT);
```

Comment:
Maybe I'm misunderstanding the question, but isn't this the minimum bound check?

Reply:
I did make some changes to Carl's "minimum bound check" behavior. It's hard to define the minimum size of the send buffer, since it's based on max fragment size, which is based on both the local connection-level preferences and the peer's preferences. So I set the minimum to the true minimum: the minimum assuming a fragment size of 1. If the send buffer ends up being too small for the configured max fragment size, we just use a smaller fragment size. This might be useful to memory-constrained customers.

Reply:
Ignore me... I read it as `<` 😂

Reply:
That being said... So we want to put an upper bound on this? Technically they could make it ~4GB

Reply:
I'm not sure it needs an upper bound. If someone really wanted a 4GB buffer, why stop them?